### ISSUE TITLE
Omit the diffusion gradient term via graph detaching instead of SpecifyGradient hack

### ISSUE THREAD

**Issue by Xallt** [ashawkey/stable-dreamfusion#335]
**Is your feature request related to a problem? Please describe.**
One issue in the repo addressed the fact that the loss graph in Tensorboard visualizes the SDS loss as a constant: https://github.com/ashawkey/stable-dreamfusion/issues/65 . In short, this happens due to the implementation of the SpecifyGradient class in `sd_utils.py`, where it returns a constant, and re-implements the gradient calculation to omit the term as in the SDS loss definition from the paper.

**Describe the solution you'd like**
While working on a fork of this repo myself, I remembered a trick that the [ScoreGAN](https://arxiv.org/abs/2305.16150) paper suggests to remove the diffusion gradient term in a similar manner. They simply detach the Tensor with `.grad` they want to ignore: https://github.com/White-Link/gpm/blob/master/gpm/models/score_gan/score_gan.py#L260
This way, the `.item()` value of the loss actually has meaning, and the gradient of this loss is the same one that we need.

So, my suggestion is just to replace
```python
w = (1 - self.alphas[t])
grad = grad_scale * w[:, None, None, None] * (noise_pred - noise)
grad = torch.nan_to_num(grad)
loss = SpecifyGradient.apply(latents, grad)
```
with
```python
w = (1 - self.alphas[t])
grad = noise_pred - noise
grad = torch.nan_to_num(grad)
loss = (grad.detach() * latents * w[:, None, None, None]).sum() *  grad_scale
```

Already checked in my experiments that this doesn't break anything. If any of the maintainers thinks this is a good idea, I'll make a pull request with a couple experiments for the latest version of the code proving that nothing breaks

**Comment by Xallt** [comment]
Actually, we get `noise_pred` from a `torch.no_grad()` block, so we may not even need to do `grad.detach()`

**Comment by ashawkey** [comment]
Yes, this is a really neat implementation! `detach()` is not needed, and we can simply change the last line to `loss = (grad * latents).sum()`.
A PR will be highly appreciated!

**Comment by ashawkey** [comment]
I have implemented it now and add your credits! Thanks for pointing this out again!

**Comment by Xallt** [comment]
@ashawkey what do you think about implementing the loss the same way as in https://github.com/threestudio-project/threestudio/issues/266 ?

**Comment by ashawkey** [comment]
Yes, I have done that!
